target/linux/ipq40xx/image/generic.mk

```diff
@@ -1117,6 +1118,26 @@ define Device/sony_ncp-hg100-cellular
 endef
 TARGET_DEVICES += sony_ncp-hg100-cellular
 
+define Device/tplink-zImage
+	DEVICE_VENDOR := TP-Link
+	IMAGES += factory.bin
+	IMAGE/factory.bin := append-rootfs | tplink-safeloader factory
+	IMAGE/sysupgrade.bin := append-rootfs | tplink-safeloader sysupgrade | append-metadata
+	KERNEL_SUFFIX := -zImage.itb
+	KERNEL = kernel-bin | fit none $(KDIR)/image-$$(DEVICE_DTS).dtb
+	KERNEL_NAME := zImage
+	SOC := qcom-ipq4019
+	TPLINK_BOARD_ID :=
+endef
+define Device/tplink_deco-m9plus-v2
+	$(call Device/tplink-zImage)
+	DEVICE_MODEL := Deco-M9Plus
+	DEVICE_VARIANT := v2
+	TPLINK_BOARD_ID := DECO-M9Plus
```

Comment:
nitpick: TPLINK_BOARD_ID is used when calling tplink-safeloader, and all its names are all caps (probably need version as well)